################################################################################
SHORT DESCRIPTION:
################################################################################
Remove IPX dependency

################################################################################
LONG DESCRIPTION:
################################################################################
In linux kernel 5.15, ipx has been removed, therefore the audit recipe can't
the ipx headers during compile time.

################################################################################
CHANGELOG
################################################################################
Chris Rogers <rogersc@ainfosec.com>

################################################################################
REMOVAL
################################################################################
Remove when we get to a newer version of audit in meta-selinux that deprecates
ipx support

################################################################################
UPSTREAM PLAN
################################################################################

################################################################################
INTERNAL DEPENDENCIES
################################################################################

################################################################################
PATCHES
################################################################################

--- a/auparse/interpret.c
+++ b/auparse/interpret.c
@@ -44,8 +44,10 @@
 #include <linux/ax25.h>
 #include <linux/atm.h>
 #include <linux/x25.h>
-#include <linux/if.h>   // FIXME: remove when ipx.h is fixed
-#include <linux/ipx.h>
+#ifdef HAVE_IPX_HEADERS
+  #include <linux/if.h>   // FIXME: remove when ipx.h is fixed
+  #include <linux/ipx.h>
+#endif
 #include <linux/capability.h>
 #include <sys/personality.h>
 #include <sys/prctl.h>
@@ -1158,7 +1160,8 @@ static const char *print_sockaddr(const
 					      x->sax25_call.ax25_call[6]);
                         }
                         break;
-                case AF_IPX:
+#ifdef HAVE_IPX_HEADERS
+		case AF_IPX:
                         {
                                 const struct sockaddr_ipx *ip =
                                                 (struct sockaddr_ipx *)saddr;
@@ -1167,6 +1170,7 @@ static const char *print_sockaddr(const
 					str, ip->sipx_port, ip->sipx_network);
                         }
                         break;
+#endif
                 case AF_ATMPVC:
                         {
                                 const struct sockaddr_atmpvc* at =
--- a/configure.ac
+++ b/configure.ac
@@ -414,6 +414,12 @@ if test x"$LIBWRAP_LIBS" != "x"; then
 	AC_DEFINE_UNQUOTED(HAVE_LIBWRAP, [], Define if tcp_wrappers support is enabled )
 fi
 
+# linux/ipx.h - deprecated in 2018
+AC_CHECK_HEADER(linux/ipx.h, ipx_headers=yes, ipx_headers=no)
+if test $ipx_headers = yes ; then
+	AC_DEFINE(HAVE_IPX_HEADERS,1,[IPX packet interpretation])
+fi
+
 # See if we want to support lower capabilities for plugins
 LIBCAP_NG_PATH
 
